conans/test/functional/graph/package_id_modes_test.py

```diff
@@ -21,10 +22,12 @@ def test_recipe_modes(self):
         configs.append((mode, "liba/1.1.1@user/stable", "b41d6c026473cffed4abded4b0eaa453497be1d2"))
 
         def _assert_recipe_mode(ref_arg, package_id_arg):
-            libb_ref = "libb/0.1@user/testing"
-            self._cache_recipe(ref, TestConanFile("liba", "0.1.1"))
-            self._cache_recipe(libb_ref, TestConanFile("libb", "0.1", requires=[ref_arg]))
-            deps_graph = self.build_graph(TestConanFile("app", "0.1", requires=[libb_ref]))
+            libb_ref = ConanFileReference.loads("libb/0.1@user/testing")
+            self._cache_recipe(ref_arg, GenConanfile().with_name("liba").with_version("0.1.1"))
```

Comment:
I am not sure but maybe there is a mistake here. Please review to verify
```suggestion
            self._cache_recipe(libb_ref, GenConanfile().with_name("liba").with_version("0.1.1"))
```

Reply:
No, it was an issue before as we were not using the `ref_arg` passed as an argument but the `ref` from the `for-loop` that calls this function. The change is `ref`->`ref_arg` (`libb_ref` will be used as a requirement later)